projects/igniteui-angular/src/lib/chips/chips-area.spec.ts

```diff
@@ -93,6 +174,576 @@ describe('IgxChipsArea', () => {
+
+        const firstChipX = (firstChipLeft + firstChipRight) / 2;
+        const firstChipY = (firstChipTop + firstChipBottom) / 2;
+
+        const secondChipTop = secondChipElem.getBoundingClientRect().top;
+        const secondChipLeft = secondChipElem.getBoundingClientRect().left;
+        const secondChipBottom = secondChipElem.getBoundingClientRect().bottom;
+        const secondChipRight = secondChipElem.getBoundingClientRect().right;
+
+        const secondChipX = (secondChipLeft + secondChipRight) / 2;
+        const secondChipY = (secondChipTop + secondChipBottom) / 2;
+
+        simulatePointerEvent('pointerdown', secondChipElem, secondChipX, secondChipY);
+        fix.detectChanges();
+
+        fix.whenStable().then(() => {
+            fix.detectChanges();
+            simulatePointerEvent('pointermove', secondChipElem, secondChipX + 10, secondChipY + 10);
+
+            return fix.whenStable();
+        }).then(() => {
+            fix.detectChanges();
+            simulatePointerEvent('pointermove', secondChip.dragDir['_dragGhost'], firstChipX, firstChipY);
+            fix.detectChanges();
+
+            return fix.whenRenderingDone();
+        }).then(() => {
+            simulatePointerEvent('pointerup', secondChip.dragDir['_dragGhost'], firstChipX, firstChipY);
+            return fix.whenRenderingDone();
+        }).then(() => {
+            setTimeout(() => {
+                const afterDropSecondChipLeft = secondChipElem.getBoundingClientRect().left;
+                expect(afterDropSecondChipLeft).toEqual(firstChipLeft);
+
+                const afterDropFirstChipLeft = firstChipElem.getBoundingClientRect().left;
+                expect(afterDropFirstChipLeft).not.toEqual(firstChipLeft);
+            });
+        });
+    }));
+
+    it('should fire onClick event', () => {
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+
+        const firstChipComp = fix.componentInstance.chips.toArray()[1];
+        spyOn(firstChipComp.onClick, 'emit');
+
+        firstChipComp.chipArea.nativeElement.dispatchEvent(new PointerEvent('pointerdown', { pointerId: 1}));
+        firstChipComp.chipArea.nativeElement.dispatchEvent(new PointerEvent('pointerup'));
+
+        fix.detectChanges();
+        expect(firstChipComp.onClick.emit).toHaveBeenCalled();
+    });
+
+    it('should fire onSelection event', () => {
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+
+        const secondChipComp = fix.componentInstance.chips.toArray()[1];
+
+        spyOn( secondChipComp.onSelection, 'emit');
+        secondChipComp.chipArea.nativeElement.focus();
+
+        const keyEvent = new KeyboardEvent('keydown', {
+            'key': ' '
+        });
+        secondChipComp.chipArea.nativeElement.dispatchEvent(keyEvent);
+        fix.detectChanges();
+        expect(secondChipComp.onSelection.emit).toHaveBeenCalled();
```

Comment:
Would be good to add unselecting by pressing again Spacebar in the same test.
Could you check the event attributes that the onSelection returns. Mainly for the array of selected chips. First it should be the chip, then should be empty when unselected.